backend/schema_migration/migrate.py

```diff
@@ -1,16 +1,73 @@
+import logging
 import os
 import sys
+from typing import Dict, List
 
 from backend.layers.business.business import BusinessLogic
+from backend.layers.business.entities import CollectionQueryFilter
 from backend.layers.persistence.persistence import DatabaseProvider
+from backend.layers.processing import logger
 from backend.layers.thirdparty.s3_provider import S3Provider
 from backend.layers.thirdparty.uri_provider import UriProvider
+from backend.portal.api.providers import get_business_logic
+
+logger.configure_logging(level=logging.INFO)
+
+
+def gather_collections() -> Dict[str, Dict[str, List[str]]]:
+    """
+    This function is used to gather all the collections and their datasets that will be migrated
+    :return: A dictionary with the following structure:
+    {
+        "published": {"collection_id": ["dataset_id", "dataset_id", ...], ...},
+        "revision": {"collection_version_id": ["dataset_id", "dataset_id", ...], ...}
+        "private": {"collection_id": ["dataset_id", "dataset_id", ...], ...},
+    }
+    """
+    businnes_logic: BusinessLogic = get_business_logic()
```

Comment:
since business logic is not stateful, we can make it a global and call it into the migrate functions. If we switch to having multiple files for container then we will want to revisit this.

Reply:
I think this complicates mocking for tests, since the global init of the provider classes would be called on import, which raises errors in our unit tests for things that require network calls (i.e. database provider)

Reply:
I think we should just make this a class with a business_logic attribute set at init. Working on this in my branch if you want to check-in about it at our sync